components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -210,14 +191,16 @@ func (src *Subscription) setV1ProtocolFields(dst *v1alpha2.Subscription) {
 
 // setV2SpecTypes sets event types in the Subscription Spec in the v1alpha2 way.
 func (src *Subscription) setV2SpecTypes(dst *v1alpha2.Subscription) error {
-	for _, filter := range src.Spec.Filter.Filters {
-		if dst.Spec.Source == "" {
-			dst.Spec.Source = filter.EventSource.Value
+	if src.Spec.Filter != nil {
+		for _, filter := range src.Spec.Filter.Filters {
+			if dst.Spec.Source == "" {
```

Comment:
I guess in case when the typeMaching equals to `exact` we don't need the require the source